### PR TITLE
Fix log preview performance and export as zip

### PR DESCRIPTION
**Pull request by mnbf9rca**
## Summary
- Add `getLogContentTail(maxLines:)` method to efficiently read only the last N lines of logs
- Update LogPreviewView to show last 1000 lines instead of loading all logs (up to 25MB)
- Add truncation notice ("Showing last 1000 lines") to preview UI
- Change log export to use zip archive instead of plain text file

## Test plan
- [x] Unit tests pass for new `getLogContentTail` method
- [ ] Verify log preview loads quickly with large log files (400KB+)
- [ ] Verify search still works in preview
- [ ] Verify export creates `.zip` file containing logs and device info

## Summary by Sourcery

Improve log viewing and exporting to handle large log files more efficiently.

New Features:
- Add support for exporting logs as a zip archive instead of a plain text file.

Enhancements:
- Add a tailing API to retrieve only the last N log lines for use in previews.
- Update the log preview UI to load only the last 1000 lines and display a truncation notice for large logs.

Tests:
- Add unit tests covering the new log tailing behavior, including limits, recency, and empty-log handling.

### CODE REVIEW

File: FamilyFoqos.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -885,7 +889,7 @@
 				CODE_SIGN_ENTITLEMENTS = foqos/foqos.entitlements;
 				CODE_SIGN_IDENTITY = "Apple Development";
 				CODE_SIGN_STYLE = Automatic;
-				CURRENT_PROJECT_VERSION = 5;
+				CURRENT_PROJECT_VERSION = 1;
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply FamilyFoqos.xcodeproj/project.pbxproj]: This is valid per Apple TN2420. For iOS apps, build numbers can reset across different marketing versions. Since MARKETING_VERSION changed from 1.31.0 → 1.31.1 (new release train), resetting CURRENT_PROJECT_VERSION is allowed.

File: Foqos/Utils/Log.swift [hunk 2]
```diff
@@ -322,6 +322,35 @@ final class Log {
...
+      } else {
+        // Take only the last neededLines from this file
+        let tailLines = Array(lines.suffix(neededLines))
+        collectedLines.insert(contentsOf: tailLines, at: 0)
+      }
+    }
+
+    // Return most recent lines (last maxLines)
+    let result = Array(collectedLines.suffix(maxLines))
+    return result.joined(separator: "\n")
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Utils/Log.swift]: Traced through the algorithm - it's correct. The method reads newest files first, prepends older content, and returns suffix(maxLines). This produces chronologically ordered output (oldest first). Added a test in a389bbd that verifies entries appear in the order they were logged.

File: Foqos/Views/LogPreviewView.swift [hunk 3]
```diff
@@ -29,11 +30,18 @@ struct LogPreviewView: View {
           )
         } else {
           ScrollView {
-            Text(filteredContent)
-              .font(.system(.caption, design: .monospaced))
-              .frame(maxWidth: .infinity, alignment: .leading)
-              .padding()
-              .textSelection(.enabled)
+            VStack(alignment: .leading, spacing: 8) {
+              Text("Showing last \(maxPreviewLines) lines")
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Views/LogPreviewView.swift]: Fixed in a389bbd - changed to "Showing up to X lines" which is accurate regardless of actual log size.

File: FamilyFoqos.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -1134,7 +1138,7 @@
 				ASSETCATALOG_COMPILER_WIDGET_BACKGROUND_COLOR_NAME = WidgetBackground;
 				CODE_SIGN_ENTITLEMENTS = FoqosWidget/FoqosWidgetExtension.entitlements;
 				CODE_SIGN_STYLE = Automatic;
-				CURRENT_PROJECT_VERSION = 5;
+				CURRENT_PROJECT_VERSION = 1;
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply FamilyFoqos.xcodeproj/project.pbxproj]: This is valid per Apple TN2420. For iOS apps, build numbers can reset across different marketing versions. Since MARKETING_VERSION changed from 1.31.0 → 1.31.1 (new release train), resetting CURRENT_PROJECT_VERSION is allowed.

File: FamilyFoqos.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -1036,7 +1040,7 @@
 			buildSettings = {
 				CODE_SIGN_ENTITLEMENTS = FoqosShieldConfig/FoqosShieldConfig.entitlements;
 				CODE_SIGN_STYLE = Automatic;
-				CURRENT_PROJECT_VERSION = 5;
+				CURRENT_PROJECT_VERSION = 1;
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply FamilyFoqos.xcodeproj/project.pbxproj]: This is valid per Apple TN2420. For iOS apps, build numbers can reset across different marketing versions. Since MARKETING_VERSION changed from 1.31.0 → 1.31.1 (new release train), resetting CURRENT_PROJECT_VERSION is allowed.

File: FamilyFoqos.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -985,12 +989,12 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				CODE_SIGN_STYLE = Automatic;
-				CURRENT_PROJECT_VERSION = 5;
+				CURRENT_PROJECT_VERSION = 1;
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply FamilyFoqos.xcodeproj/project.pbxproj]: This is valid per Apple TN2420. For iOS apps, build numbers can reset across different marketing versions. Since MARKETING_VERSION changed from 1.31.0 → 1.31.1 (new release train), resetting CURRENT_PROJECT_VERSION is allowed.

File: FamilyFoqos.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -934,12 +938,12 @@
 			buildSettings = {
 				BUNDLE_LOADER = "$(TEST_HOST)";
 				CODE_SIGN_STYLE = Automatic;
-				CURRENT_PROJECT_VERSION = 5;
+				CURRENT_PROJECT_VERSION = 1;
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply FamilyFoqos.xcodeproj/project.pbxproj]: This is valid per Apple TN2420. For iOS apps, build numbers can reset across different marketing versions. Since MARKETING_VERSION changed from 1.31.0 → 1.31.1 (new release train), resetting CURRENT_PROJECT_VERSION is allowed.

File: FamilyFoqos.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -1101,7 +1105,7 @@
 				ASSETCATALOG_COMPILER_WIDGET_BACKGROUND_COLOR_NAME = WidgetBackground;
 				CODE_SIGN_ENTITLEMENTS = FoqosWidget/FoqosWidgetExtension.entitlements;
 				CODE_SIGN_STYLE = Automatic;
-				CURRENT_PROJECT_VERSION = 5;
+				CURRENT_PROJECT_VERSION = 1;
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply FamilyFoqos.xcodeproj/project.pbxproj]: This is valid per Apple TN2420. For iOS apps, build numbers can reset across different marketing versions. Since MARKETING_VERSION changed from 1.31.0 → 1.31.1 (new release train), resetting CURRENT_PROJECT_VERSION is allowed.
